hack/e2e.sh

```diff
@@ -26,6 +23,35 @@ E2E_KUTTL_TEST=${E2E_KUTTL_TEST:-}
 # Define a custom external network
 export E2E_EXTERNAL_NETWORK_NAME=${E2E_EXTERNAL_NETWORK_NAME:-private}
 
+creds_dir=$(mktemp -d)
+
+function logresources() {
+    # Log all resources if exiting with an error
+    if [ $? != 0 ]; then
+        kubectl get openstack -o yaml -A
+    fi
+}
+
+function cleanup() {
+    rm -rf -- "$creds_dir"
+    logresources
```

Comment:
You need to either flip these round or combine them in a single function. logresources uses `$?`, which will be overwritten by running `rm`.